src/util/chunked_content_decoder.js

```diff
@@ -8,27 +8,50 @@ const STATE_WAIT_NL_HEADER = 'STATE_WAIT_NL_HEADER';
 const STATE_SEND_DATA = 'STATE_SEND_DATA';
 const STATE_WAIT_CR_DATA = 'STATE_WAIT_CR_DATA';
 const STATE_WAIT_NL_DATA = 'STATE_WAIT_NL_DATA';
+const STATE_READ_TRAILER = 'STATE_READ_TRAILER';
+const STATE_WAIT_NL_TRAILER = 'STATE_WAIT_NL_TRAILER';
+const STATE_WAIT_NL_END = 'STATE_WAIT_NL_END';
 const STATE_CONTENT_END = 'STATE_CONTENT_END';
 const STATE_ERROR = 'STATE_ERROR';
 
 const CR_CODE = '\r'.charCodeAt(0);
 const NL_CODE = '\n'.charCodeAt(0);
 
+// lenient limits to avoid abuse
+const MAX_CHUNK_SIZE = 1024 * 1024 * 1024 * 1024;
+const MAX_CHUNK_HEADER_SIZE = 1024;
+const MAX_TRAILER_SIZE = 1024;
+const MAX_TRAILERS = 20;
+
 /**
  *
  * ChunkedContentDecoder
  *
  * Take a data stream and removes chunking signatures from it
  *
+ * Basic encoding structure:
+ * ---------------------------------------------------
+ * 1fff;chunk-signature=1a2b\r\n   - chunk header
```

Comment:
Not sure I fully understand this structure, where can we see the size of the chunk data here? is the `fff` is part of the structure?

Reply:
1fff is the size

Reply:
see https://en.wikipedia.org/wiki/Chunked_transfer_encoding -
> If chunk extensions are provided, the chunk size is terminated by a semicolon and followed by the parameters, each also delimited by semicolons. Each parameter is encoded as an extension name followed by an optional equal sign and value. These parameters could be used for a running [message digest](https://en.wikipedia.org/wiki/Message_digest) or [digital signature](https://en.wikipedia.org/wiki/Digital_signature), or to indicate an estimated transfer progress, for instance.

Reply:
I added a comment about it, hope it is clearer.